### PR TITLE
chore: Add Zenodo DOI to official project

### PR DESCRIPTION
**Pull request by uwcdc**
This PR creates a `CITATION.cff` and adds a DOI badge to the `README` file.

Closes PR https://github.com/uw-ssec/caustics/issues/52

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,25 @@
+cff-version: 1.0.0
+message: "If you use this software, please cite it as below."
+authors:
+- family-names: "Stone"
+  given-names: "Connor"
+  orcid: "https://orcid.org/0000-000x-xxxx-xxxx"
```
Comment (ConnorStoneAstro) [inline CITATION.cff:6]: My ORCID is 0000-0002-9086-6398

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,25 @@
+cff-version: 1.0.0
+message: "If you use this software, please cite it as below."
+authors:
```
Comment (ConnorStoneAstro) [inline CITATION.cff:3]: I think We'll want everyone who's a contributor listed here. We can fill the ORCID later. So it should also include:
- Adam Coogan
- Andreas Filipp
- Charles Wilson
- Misha Yantovski-Barth
- Ronan Legin

**Review by ConnorStoneAstro — CHANGES_REQUESTED** [review]

**Review by ConnorStoneAstro — APPROVED** [review]

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+- family-names: "Wilson"
+  given-names: "Charles"
+- family-names: "Yantovski-Barth"
+  given-names: "Misha"
+- family-names: "Legin"
+  given-names: "Ronan"
+title: "caustics"
+doi: 10.5281/zenodo.10806383
+version: v0.9.0
+date-released: 2024-03-11
```
Comment (lsetiawan) [inline CITATION.cff:35]:
```suggestion
# Update below with the appropriate
# zenodo DOI, version, and release date
# for each release
doi: 10.5281/zenodo.10806383
version: v0.9.0
date-released: 2024-03-11
```

Comment (ConnorStoneAstro) [reply CITATION.cff:35]: Will we need to update with every release? Is there a way to automate this?

Comment (lsetiawan) [reply CITATION.cff:35]:
```suggestion
doi: 10.5281/zenodo.10806382
abstract: "The lensing pipeline of the future: GPU-accelerated, automatically-differentiable, highly modular. Currently under heavy development: expect interface changes and some imprecise/untested calculations."
repository-code: "https://github.com/Ciela-Institute/caustics"
```

If you want to automate it, then let's not specify a version, but rather cite for all versions

Comment (lsetiawan) [reply CITATION.cff:38]: @uwcdc please modify to the second suggestion above

Comment (lsetiawan) [reply CITATION.cff:38]: Also, remove the comments

**Review by lsetiawan — CHANGES_REQUESTED** [review]
Looks like this need to be modified to use the DOI for all versions rather than a specific one.

**Comment by uwcdc** [comment]
@lsetiawan All of the DOI information should reflect the `10.5281/zenodo.10806382` (all version DOI) in the citation and README

**Pull request merged**